cEP-0028.md

```diff
@@ -0,0 +1,428 @@
+   This extension of support for `tags` would require at least the
+   following changes:
+
+   - `coalib.settings.Section`<br/>
+     Adds a new method to the Section class to check if
+     the section should be enabled for a particular tag
+     based invocation.
+
+   ```python
+   class Section:
+
+     def is_enabled_for_tags(self, enabled_tags):
+       """
+       Checks if this section is enabled when the execution
+       limits are specified by tags.
+
+       :param enabled_tags: List of all enabled tags.
+       :return:             True or False
+       """
+       if len(enabled_tags) == 0:
+         return True
+
+       set_enabled = set(enabled_tags)
+       set_tags = set(self.get('tags', []))
+
+       matching = set_tags.intersection(set_enabled)
+       return len(matching) > 0
+   ```
+
+   - `coalib.parsing.DefaultArgParser.default_arg_parser`<br/>
+     Add a new `tags` cli argument type to the argparser.
+
+     ```python
+     misc_group.add_argument(
+         '-t', '--tags', nargs='*', type=str,
+         help='tags of sections to execute')
+     ```
+
+   - `coalib.coala_main.run_coala`<br/>
+     Patch `run_coala()` to check if the corresponding tags
+     of the section are enabled.
+
+   ### Supported Tags
+
+   The following are the supported default tags groups:
+
+   - `open`: Tag group to execute during a open event on a file.
+
+   - `change`: Tag group to execute during a change event on the
+               project files.
+
+   - `save`: Tag group to execute on a save event on the project files.
+
+   - `formatting`: Tag group to execute when a `formatting` event is
+     raised on a file.
+
+2. Custom entry point into coala core<br/>
+   Since coala core was not designed to be a long running process coalib
+   does not provide a flexible entry point into it. To maximize the
+   efficiency of the coala language server, it will require an entry
+   point that supports:
+
+   - Support for `FileProxies`: Since not all changes on the editor are
+     flushed to the disk coala core will have to support processing file
+     content directly as opposed to files.
+
+   The following functions are core to the implementation and API of
+   coala and need to be patch or extended to bring extended support
+   for coala-ls.
+
+   - `coalib.io.FileProxy`<br/>
+     Addition of a FileProxy class.
+
+   ```python
+   class FileProxy:
+     """
+     coala requires the files to be flushed to perform
+     analysis on. This provides an alternative by providing
+     an always updated proxy of the said file by watching
+     for events from the client such as didChange.
+     """
+
+     @classmethod
+     def from_name(cls, file, workspace):
+       """
+       Construct a FileProxy instance from an existing
+       file on the drive.
+
+       :param file:
+           The name of the file to be represented by
+           the proxy instance.
+       :param workspace:
```

Comment:
Will this be a string ?  Or can it be something else?  Use arg annotations to help `file : str, workspace: str`